### PR TITLE
fix: in oai chat completion, use msg name field instead of prepended to msg content

### PR DESCRIPTION
**Pull request by stevenksmith**
supersedes #751
addresses part of #739

### CODE REVIEW

File: srv/adapter/chat-completion.ts [hunk 1]
```diff
@@ -249,15 +249,23 @@ export async function toChatCompletionPayload(
...
-    const obj: CompletionItem = {
-      role: 'assistant',
-      content: line.trim().replace(BOT_REPLACE, replyAs.name).replace(SELF_REPLACE, handle),
-    }
-
     const isSystem = line.startsWith('System:')
     const isUser = line.startsWith(handle)
     const isBot = !isUser && !isSystem
 
+    const name = isBot ? replyAs.name : isUser ? handle : undefined
```
Comment (khanonnie) [inline srv/adapter/chat-completion.ts]: Is `handle` by this point accounting for impersonated characters? I'd assume so

**Review by khanonnie — APPROVED** [review]
i like this more than having an entirely new option for this behavior, since not including char/user names in messages at all seemed to make outputs worse so i don't think you'd ever want to turn it off.

File: srv/adapter/chat-completion.ts (hunk 1 above)
Comment (stevenksmith) [reply srv/adapter/chat-completion.ts]: Good catch, this was breaking past impersonations (registering them as the main character in assistant message), and example messages (incorrect splitting). This is fixed in c4eefd5

**Comment by stevenksmith** [comment]
Sorry for flipflopping but I now think #751 is better.

**Comment by sceuick** [comment]
> Sorry for flipflopping but I now think #751 is better.

Yes, I think this will have bad side effects.

**Pull request closed without merging**